examples/ecs/run_conditions.rs

```diff
@@ -0,0 +1,98 @@
+//! This example demonstrates how to use run conditions to control when systems run.
+
+use bevy::prelude::*;
+
+fn main() {
+    println!();
+    println!("For the first 2 seconds you will not be able to increment the counter");
+    println!("Once that time has passed you can press space, enter, left mouse, right mouse or touch the screen to increment the counter");
+    println!();
+
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .init_resource::<InputCounter>()
+        .add_system(
+            increment_input_counter
+                // The common_conditions module has a few useful run conditions
+                // for checking resources and states, these are included in prelude
+                .run_if(resource_exists::<InputCounter>())
+                // This is our custom run condition, both this and the
+                // above condition must be true for the system to run
+                .run_if(has_user_input),
+        )
+        .add_system(
+            print_input_counter
+                // This is also a custom run condition but this time in the form of a closure,
+                // this is useful for small, simple run conditions you don't need to reuse.
+                // All the normal rules still apply, all parameters must be read only except for local parameters
+                // In this case we will only run if the input counter resource exists and has changed but not just been added
+                .run_if(|res: Option<Res<InputCounter>>| {
+                    if let Some(counter) = res {
+                        counter.is_changed() && !counter.is_added()
+                    } else {
+                        false
+                    }
+                }),
+        )
+        .add_system(
+            print_time_message
+                // This is a custom generator function that returns a run
+                // condition, must like the common conditions module.
+                // It will only return true once 2 seconds has passed
```

Comment:
```suggestion
                // This function returns a custom run condition, much like the common conditions module.
                // It will only return true once 2 seconds have passed.
```